src/store.js

```diff
@@ -28,6 +28,10 @@ export default class Store {
     if (this.prevState.username && !this.state.username) {
       this.state.username = this.prevState.username;
     }
+    if (this.state.status === 'EXTERNAL_AUTHENTICATION_COMPLETED') {
+      await this.dispatch('POST_FLOW', 'continueAuthentication', '{}');
```

Comment:
I'm not sure I understand this. We are hijacking a different dispatch call and replacing it with a continueAuthentication?

What is this needed for?

Reply:
yeah...I just want it to automatically `continueAuthentication` when the state is 'EXTERNAL_AUTHENTICATION_COMPLETED'. Is there's a better place you think this sound go?

Reply:
I like anything that is based on a specific state to go in the handlebar. I think you could move it to there where is submits a hidden button after a few seconds. It gives you a nice chance to display a nice big green Success message right before moving on as well.

@cindytu-ping indicated there were issues with adding this to the template. Cindy can you elaborate on what the issue was here.

I'm afraid that we are going to have index/store.js be very bloated with specific state code and it will become harder to maintain over time.

Reply:
sure, I can look into getting it to do that instead. I was just thinking it would be nice to have it just do the POST automatically without actually showing anything to the user.

Reply:
I wasn't able to add this to the template because I couldn't find a good way to have the hidden button submitted after a few seconds, some methods I found online suggests using the body tag which is not a part of the handlebars.

What I ended up doing is https://github.com/pingidentity/pf-authn-js-widget/blob/588b154b2075b867d647a09e38e9c0b656c16d48/src/index.js#L180. I called dispatch for the specific state inside of a function in index.js.

Reply:
Yeah, I much prefer this approach of adding handlers for specific states. Following the approach of adding an event handler is nice.

I see now the issue that we can't execute handlebar javascript just by the nature of how handlebars are rendered

If we need some common functionality for states I think we need to add hooks for those states to do what they need to do.

As an example we can add a [post render](https://github.com/pingidentity/pf-authn-js-widget/commit/82668e9a33f351f251d2614bd84db683f95645f5) callback that gets called when a specific state is rendered. Example of this is [here](https://github.com/pingidentity/pf-authn-js-widget/commit/9032faa9b1aae55a45c7f0507567cd9bafbddcb8). The example just pops up an alert 2 seconds after the template is rendered. 

If this can't work with the current addEventListener, maybe we should pull in the post-render hook or something like it.